log_test.go

```diff
@@ -1044,3 +1010,23 @@ func mustTerm(term uint64, err error) uint64 {
 	}
 	return term
 }
+
+type index uint64
```

Comment:
Idea of having clearer test case generations seems good, however I'm not sure about the solution you proposed, it doesn't seem very natural. 

Things like `index(0).terms(1,2,3)` that generate pairs `(index, term)` of `(0, 1), (1, 2), (2, 3)`. Also having a function as part of `to` method seems like you really want to force a single function to handle too many different cases.

It's nice that names are short and concise, but downside is that they don't represent what code really does.

`index` seems like more `fromIndex` or `index.from`. `to` seems like it would better without function and depend on terms provided via a `terms`.

For example replace `index(0).to(2, func(index uint64) uint64 { return index })` with `index.from(0).terms(term.from(0).to(2)...)`

Reply:
Good point re `func` generator in `to` being ugly. Replaced it with a simple `index(0).termsIn(from, to)`.

So we now have 2 generators:

```golang
index(1).terms(1, 2, 2, 3) // the terms are enumerated explicitly
index(1).termsIn(100, 200) // consecutive terms [100, 200)
```

I like the conciseness of `index` better than `fromIndex`, it's kinda the point - for writing less boilerplate in tests. I fixed a slight obscurity of this name by adding some comments to all generators.

Reply:
> Good point re func generator in to being ugly. Replaced it with a simple index(0).termsIn(from, to).

Nice, however it would be nicer to follow etcd name convention `index(0).termRange(from, to)`

Reply:
Done.